custom-routes-vis/src/main/webapp/script.js

```diff
@@ -335,15 +335,56 @@ function selectRouteDisplayDetails(
   let routeInfoElement = document.getElementById('route-info');
   routeInfoElement.innerText = 'Selected Route Info:' +
       '\nDuration: ' + formatDuration(totalDurationSec) +
-      '\nDistance: ' + formatDistance(totalDistanceMeters);
+      '\nDistance: ' + formatDistance(totalDistanceMeters) + '\n';
 
   if (routeToken !== '') {
-    routeInfoElement.innerText += '\nRoute Token: check the developer console';
+    let copyBtn = document.createElement('button');
+    copyBtn.innerText = 'Copy Route Token';
+    copyBtn.addEventListener('click', function() {
+      copyTokenToClipBoard(routeToken);
+    });
+    routeInfoElement.appendChild(copyBtn);
```

Comment:
Double checking that if the users generates routes multiple times the button is replaced each time, and they don't get like 50 buttons?

Have you considered creating a button and just hiding it or showing is as necessary?

Reply:
+1 about hiding the button if there is no route token.

Reply:
Yes the button is replaced when another route is generated and yes the button is hidden if there is no route token.